include/algorithms/kmeans/kmeans_types.h

```diff
@@ -90,11 +90,10 @@ enum MasterInputId
  */
 enum PartialResultId
 {
-    nObservations,                                  /*!< Table containing the number of observations assigned to centroids */
-    partialSums,                                    /*!< Table containing the sum of observations assigned to centroids */
-    partialObjectiveFunction,                       /*!< Table containing an objective function value */
-    partialGoalFunction = partialObjectiveFunction, /*!< Table containing an objective function value \DAAL_DEPRECATED */
```

Comment:
1) some problem with static BC: not defined symbols.
2) problem with dynamic BC:  user will get silently assignments instead of goal-function

Reply:
and as number of field was reduced when user requests 'partialCandidatesCentroids' with new daal.so he can get segmentation fault.

Reply:
> 1. some problem with static BC: not defined symbols.


Do you mean if the user wants to use interfaces1? 

> 2. problem with dynamic BC: user will get silently assignments instead of goal-function.

No. I deleted the alias, not the extra field.

```c++
partialObjectiveFunction,
partialGoalFunction = partialObjectiveFunction,
partialAssignments
```
```
partialObjectiveFunction = 2
partialGoalFunction = 2
partialAssignments = 3
```

Where is the problem here?

Reply:
Sounds reasonable. Thanks for clarification!
And at last could you check additionally that `lastResultI` and `lastPartialResultId` were not changed after changes?